app/controllers/application_controller/flash_notices.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+# see application_controller.rb
+module ApplicationController::FlashNotices
+  def self.included(base)
+    base.helper_method(
+      :flash_notices?, :flash_get_notices, :flash_notice_level,
+      :flash_clear, :flash_notice, :flash_warning, :flash_error
+    )
+  end
+
+  ##############################################################################
+  #
+  #  :section: Error handling
+  #
+  #  NOTE: MO doesn't use built-in Rails `flash`, i.e. session[:flash].
```

Comment:
@mo-nathan, @pellaea: I didn't realize that we rolled our own flash. 
I'm curious. Is there an short explanation?

Reply:
@JoeCohen I was surprised too, that's my note I believe, from a few years ago. 

After looking into this vs [the documentation](https://api.rubyonrails.org/classes/ActionDispatch/Flash.html), my impression was that the main difference between this and the "Rails way" is the naming of the session attribute itself, otherwise I remember feeling that as far as the Rails flash docs provide guidance, it's mostly roll your own anyway, and seems to work pretty much the same.  

But you may notice differences I missed.